tests/ci/run_valgrind_tests.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash -xu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+################################################################################
+# set -e is disabled in this script because we need the script to continue
+# even if |run_valgrind| returns an error.
```

Comment:
Curious why the `continue` is preferred here when |run_valgrind| returns an error?

Reply:
This concern has been addressed in `all_test.go`.
https://github.com/awslabs/aws-lc/blob/main/util/all_tests.go#L454-L500

Also, SIM CryptoAlg-597 mentioned current sanitizer executes all tests. Why only `crypto_test` is selected?

I have concern on current implementation because it creates a new path/start point. The existing entries are `all_tests.go` and `ssl/test/runner`.
BoringSSL has the test entry points specified in build doc, and adds some env variables to enable/disable diff flavors.
https://github.com/awslabs/aws-lc/blob/main/BUILDING.md#running-tests

Reply:
It is not a concern, it is just the logic needed to make the script behave like we want.

I don't see why this is different from how else we have implemented our test execution: through bash scripts (`run_posix_sanitizers.sh`, `run_posix_tests.sh`, `run_s2n_integration.sh`). If we wanted to unify everything through the Go script, why didn't we start that long ago and follow the same pattern for the previous tests?

Reply:
`run_s2n_integration.sh` is another story.
`run_posix_sanitizers.sh` and `run_posix_tests.sh`  execute `run_tests` target, which include `all_tests.go`. These two scripts only enables some flags. The actual compiler flags are included in the `CMakeLists.txt`.
https://github.com/awslabs/aws-lc/blob/main/CMakeLists.txt#L655-L664

> I envision adding other target executables

This maybe another reason why `all_tests.go` should be considered. When more and more executables are included, tests may need to be executed concurrently. The `all_tests.go` has implemented `worker` and `shards` to reduce the time.

Besides why only crypto_test is selected? Based on the SIM, seems all tests are required.

> If we wanted to unify everything through the Go script

Unify is not purpose, the question is about why existing Valgrind config cannot be reused.